cmd/orderimports/orderimports.go

```diff
@@ -0,0 +1,226 @@
+
+var (
+	regex      = flag.String("include-path", "cmd/kubeadm", "only files with paths matching this regex is touched")
+	ignore     = flag.String("ignore-file", "zz_generated", "files matching this regex is ignored")
+	isTerminal = term.IsTerminal(int(os.Stdout.Fd()))
+	logPrefix  = ""
+)
+
+type analyzer struct {
+	fset      *token.FileSet // positions are relative to fset
+	ctx       build.Context
+	failed    bool
+	donePaths map[string]interface{}
+	ignore    *regexp.Regexp
+}
+
+func newAnalyzer() *analyzer {
+	ctx := build.Default
+	ctx.CgoEnabled = true
+
+	ignore, err := regexp.Compile(*ignore)
+	if err != nil {
+		log.Fatalf("Error compiling ignore regex: %v", err)
+	}
+
+	a := &analyzer{
+		fset:      token.NewFileSet(),
+		ctx:       ctx,
+		donePaths: make(map[string]interface{}),
+		ignore:    ignore,
+	}
+
+	return a
+}
+
+// collect extracts test metadata from a file.
+func (a *analyzer) collect(dir string) {
+	if _, ok := a.donePaths[dir]; ok {
+		return
+	}
+	a.donePaths[dir] = nil
+
+	// Create the AST by parsing src.
+	fs, err := parser.ParseDir(a.fset, dir, nil, parser.AllErrors|parser.ParseComments)
+
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "ERROR(syntax)", logPrefix, err)
+		a.failed = true
+		return
+	}
+
+	for _, p := range fs {
+		// returns first error, but a.handleError deals with it
+		files := a.filterFiles(p.Files)
+		for _, file := range files {
+			pathToFile := a.fset.File(file.Pos()).Name()
+
+			var originalImports []string
+			var stdlibImports []string
+			var localImports []string
+			var k8sImports []string
+			var externalImports []string
+
+			for _, imp := range file.Imports {
+				importPath := strings.Replace(imp.Path.Value, "\"", "", -1)
+				parts := strings.Split(importPath, "/")
+				originalImports = append(originalImports, importPath)
+
+				if !strings.Contains(parts[0], ".") {
+					// standard library imports are first
+					stdlibImports = append(stdlibImports, importPath)
+				} else if strings.HasPrefix(importPath, "k8s.io/kubernetes") {
+					// local imports are second
+					localImports = append(localImports, importPath)
+				} else if strings.Contains(parts[0], "k8s.io") {
+					// other *.k8s.io imports are third
+					k8sImports = append(k8sImports, importPath)
+				} else {
+					// external repositories are fourth
+					externalImports = append(externalImports, importPath)
+				}
+
+			}
+			if len(originalImports) == 0 {
+				continue
+			}
+			sort.Strings(stdlibImports)
+			sort.Strings(localImports)
+			sort.Strings(k8sImports)
+			sort.Strings(externalImports)
+
+			orderImports := []string{}
+			orderImports = append(orderImports, stdlibImports...)
+			orderImports = append(orderImports, localImports...)
+			orderImports = append(orderImports, k8sImports...)
+			orderImports = append(orderImports, externalImports...)
+			if diff := cmp.Diff(originalImports, orderImports); diff != "" {
+				a.failed = true
+				fmt.Printf("%s (-got +want):\n%s", pathToFile, diff)
+			}
```

Comment:
Is s including new lines separating the import sections?

Reply:
```
fmt
                                        // <---- new line here
k8s.io/kubernetes/....
k8s.io/kubernetes/....
                                        // <---- new line here
k8s.io/
```

Reply:
> Is this including new lines separating the import sections? 


Not yet, because the `originalImports` has no blank lines, I need to see how to add blank lines to the `originalImports`